src/world.cpp

```diff
@@ -64,29 +68,46 @@ World::World(const std::string& biome_name, MapTile_t tile_type, size_t seed) :
         3, 3, macro_tile_size, height, seed, biome_, get_test_map(tile_type)
     ) {}
 
+void
+World::mark_for_update(ChunkIndex chunk_pos) {
+    chunks_to_update_.insert(chunk_pos);
+}
+
+void
+World::mark_for_update(TerrainDim3 tile_sop) {
+    if (!terrain_main_.in_range(tile_sop))
+        return;
+    Dim chunk_pos = terrain_main_.get_chunk_from_tile(tile_sop);
+    mark_for_update(chunk_pos);
+}
+
+// Should not be called except by lambda function
 __attribute__((optimize(2))) void
 World::update_single_mesh(ChunkIndex chunk_pos) {
-    const auto& chunks = terrain_main_.get_chunks();
-    // entity::Mesh chunk_mesh = entity::generate_mesh(chunks[chunk_pos]);
+    const auto& chunk = terrain_main_.get_chunk(chunk_pos);
     entity::Mesh chunk_mesh =
-        entity::ambient_occlusion_mesher(terrain::ChunkData(chunks[chunk_pos]));
+        entity::ambient_occlusion_mesher(terrain::ChunkData(chunk));
 
     chunk_mesh.change_color_indexing(
         biome_.get_materials(), terrain::TerrainColorMapping::get_colors_inverse_map()
     );
 
-    chunks_mesh_[chunk_pos]->update(chunk_mesh);
-    chunks_mesh_[chunk_pos]->set_color_texture(
-        terrain::TerrainColorMapping::get_color_texture()
-    );
+    std::scoped_lock lock(meshes_to_update_mutex_);
+    meshes_to_update_.insert({chunk_pos, chunk_mesh});
```

Comment:
if you wait below you can just modify the mesh in here

Reply:
No. I need to write a big doc string about how this works, but the function you are thinking about is an opengl call. This function is build to run in the thread pool, and can therefore not call the opengl mesh update function.

Reply:
ohh ok

add the docstring